crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -654,4 +672,38 @@ pub mod tests {
 
         assert!(encoder0.merge(encoder1).is_err());
     }
+
+    #[test]
+    fn encoding_strips_unnecessary_tokens_and_prices() {
+        let prices = hashmap! {token(1) => 7.into(), token(2) => 2.into(),
+        token(3) => 9.into(), token(4) => 44.into()};
+
+        let mut encoder = SettlementEncoder::new(prices);
+
+        let order_1_2 = OrderBuilder::default()
+            .with_sell_token(token(1))
+            .with_sell_amount(33.into())
+            .with_buy_token(token(2))
```

Comment:
nit: Note that trades also use indices for the token addresses. Could you maybe make this a `token(3)` so that in theory the token indices for the trade would change by optimizing the token array/price vector?

Reply:
Done 👍